Project/OOP/Cylinder

```diff
@@ -1 +1,30 @@
-
+class Cylinder:
+    @staticmethod
+    def make_area(d, h):
+        circle = pi * d ** 2 / 4
+        side = pi * d * h
+        return round(circle*2 + side, 2)
+    
+    def __init__(self, diameter, high):
+        self.dia = diameter
+        self.high = high
+        self.area = self.make_area(self.dia,self.high)
+    
+    def __setattr__(self, name, value):
+        if name == 'dia' or name == 'high':
+            self.__dict__[name] = value
+            self.area = self.make_area(self.dia,self.high)
```

Comment:
there's a corner case here init: when you set `dia`, but there's no `high` ( in constructor ), that will not find an object to calculate the area. so, what you can do is use super().__setattr__ in __init__.